### DISCUSSION TITLE
IjwHost on Linux?

### DISCUSSION THREAD

**Discussion by Moonfish** [dotnet/runtime#66117] [Q&A] [answered]
I have an existing managed C++ library that I'd like to try using on Linux (native app using this managed C++ library). 
On Windows, the native host is currently relying on ijwhost.dll to bind to the .NET runtime when this managed C++ library is loaded.
As IjwHost isn't included in the Linux variety of the .NET SDK, what are my options?  Is a shared/centralized .NET runtime installation on Linux even a thing?

**Comment by teo-tsirpanis** [comment, marked as answer]
Hello @Moonfish, C++/CLI is a Windows-only technology and not recommnded to build new code on top of.

The best option is to port the native library to C# (newer language features like function pointers will make this easier), and if that is not feasible, you can make the C++ library entirely native and design a communication layer between the native and managed parts of your app through P/Invokes.

You can discuss more about it on https://developercommunity.visualstudio.com/t/netcore-ccli-for-linux-and-macos/873014.
